internal/pkg/cli/task_run.go

```diff
@@ -572,14 +602,20 @@ func (o *runTaskOpts) Ask() error {
 	if o.generateCommandTarget != "" {
 		return nil
 	}
+
 	if o.shouldPromptForAppEnv() {
 		if err := o.askAppName(); err != nil {
 			return err
 		}
 		if err := o.askEnvName(); err != nil {
 			return err
 		}
+	} else if o.envFile != "" {
+		if err := o.askAppName(); err != nil {
+			return fmt.Errorf("no S3 bucket available to hold environment file %s specified in --%s: %w", o.envFile, envFileFlag, err)
+		}
```

Comment:
Should we immediately error here instead of asking for the app name?

My understanding from reading is that if the user specifies `task run --default --env-file .env` we will ignore that they provided `--default` and ask for the app name. That might be surprising for folks:

```go
return fmt.Errorf("--%s is only supported for tasks that can run in an environment with --%s and --%s", envFileFlag, appFlag, envFlag)
```

Reply:
+1

Reply:
That error message might be unintuitive since if you don't specify `--app` and `--env` it'll ask anyways - how about something like:

```go
return fmt.Errorf("--%s is not compatible with --%s because it requires an S3 bucket provided from an app & environment", envFileFlag, defaultFlag)
```

Or, should I just use `cmd.MarkFlagsMutuallyExclusive(envFileFlag, defaultFlag)`?

Reply:
> Or, should I just use cmd.MarkFlagsMutuallyExclusive(envFileFlag, defaultFlag)?

sgtm!